csharp/ql/src/API Abuse/NoDisposeCallOnLocalIDisposable.ql

```diff
@@ -81,6 +81,8 @@ private class Conf extends DataFlow::Configuration {
 
   override predicate isAdditionalFlowStep(DataFlow::Node node1, DataFlow::Node node2) {
     node2.asExpr() = any(LocalScopeDisposableCreation other | other.getAnArgument() = node1.asExpr())
+    or
+    exists(NullCoalescingExpr nce | node1.asExpr() = nce.getRightOperand() and node2.asExpr() = nce)
```

Comment:
I am surprised that we don't already have data flow steps from both operands of `??` to the `??` expression itself. I think we should instead add those steps to `LocalFlow::LocalExprStepConfiguration::candidate`.

Reply:
I completely agree, I just thought there was a subtlety in the query that I had somehow missed. That is *very* surprising. I'll make that change.